rec: Drop queries truncated because they were larger than our buffer

rgacogne opened the pull request:
### Short description
<!-- Write a small description of what this Pull Request fixes or provides, including the issue #s -->
Incoming queries larger than our buffer (1500 bytes) are truncated by the kernel, so let's not even try to parse them.

### Checklist
<!-- please indicate if any of these things are done/included with this Pull Request. Not all boxes need to be checked for the Pull Request to be accepted -->
I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [x] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

pieterlexis commented:
in #6281, I lowered the advertised maximum to 512 (minimum allowed by EDNS), perhaps the `data.resize()` should mirror this

pieterlexis commented:
@ahupowerdns will know. My feeling is "something something ethernet"

rgacogne commented:
Alright, thanks! So does anyone see a reason why we shouldn't go with `512` from now on?

Habbie commented:
> Alright, thanks! So does anyone see a reason why we shouldn't go with 512 from now on?

For rec, 512 has my full support. 

(Auth, and by extension, dnsdist may need to deal with big UPDATE messages, I suppose.)

pieterlexis commented:
> So does anyone see a reason why we shouldn't go with 512 from now on?

For the recursor? No.

rgacogne commented:
- Updated to switch to & 512-byte buffer for incoming queries
- Fixed an issue where the buffer could have been invalidated between queries because of the `resize()` call, since nothing guarantees a reallocation won't happen when the size is shrank
- Rebased because it conflicted with one commit from #6518

rgacogne commented:
Rebased to fix a conflict.
